evm_arithmetization/src/fixed_recursive_verifier.rs

```diff
@@ -1350,97 +1351,101 @@ where
+                        Table::all()[table],
+                        original_degree_bits,
+                    ))
+                })?
+                .shrink(stark_proof, &all_proof.multi_proof.ctl_challenges)?;
+            let index_verifier_data = table_circuits
+                .by_stark_size
+                .keys()
+                .position(|&size| size == original_degree_bits)
+                .unwrap();
+            root_inputs.set_target(
+                self.root.index_verifier_data[table],
+                F::from_canonical_usize(index_verifier_data),
             );
+            root_inputs.set_proof_with_pis_target(&self.root.proof_with_pis[table], &shrunk_proof);
 
-            set_public_value_targets(
-                &mut root_inputs,
-                &self.root.public_values,
-                &all_proof.public_values,
-            )
-            .map_err(|_| {
-                anyhow::Error::msg("Invalid conversion when setting public values targets.")
-            })?;
+            check_abort_signal(abort_signal.clone())?;
+        }
 
-            let root_proof = self.root.circuit.prove(root_inputs)?;
+        root_inputs.set_verifier_data_target(
+            &self.root.cyclic_vk,
+            &self.segment_aggregation.circuit.verifier_only,
+        );
 
-            Ok(Some(ProverOutputData {
-                proof_with_pis: root_proof,
-                public_values: all_proof.public_values,
-            }))
-        } else {
-            Ok(None)
-        }
+        set_public_value_targets(
+            &mut root_inputs,
+            &self.root.public_values,
+            &all_proof.public_values,
+        )
+        .map_err(|_| {
+            anyhow::Error::msg("Invalid conversion when setting public values targets.")
+        })?;
+
+        let root_proof = self.root.circuit.prove(root_inputs)?;
+
+        Ok(ProverOutputData {
+            proof_with_pis: root_proof,
+            public_values: all_proof.public_values,
+        })
     }
 
+    /// Returns a proof for each segment that is part of a full transaction
+    /// proof.
     pub fn prove_all_segments(
         &self,
         all_stark: &AllStark<F, D>,
         config: &StarkConfig,
         generation_inputs: GenerationInputs,
-        max_cpu_len: usize,
+        max_cpu_len_log: usize,
         timing: &mut TimingTree,
         abort_signal: Option<Arc<AtomicBool>>,
     ) -> anyhow::Result<Vec<ProverOutputData<F, C, D>>> {
-        let mut segment_index = 0;
-        let mut proofs = vec![];
-        while let Some(proof) = self.prove_segment(
-            all_stark,
-            config,
-            generation_inputs.clone(),
-            max_cpu_len,
-            segment_index,
-            timing,
-            abort_signal.clone(),
-        )? {
-            segment_index += 1;
+        let mut all_data_segments =
+            generate_all_data_segments::<F>(Some(max_cpu_len_log), generation_inputs.clone())?;
```

Comment:
This method implies having a single process handle the entire proving. Is the API flexible enough to allow zero-bin (or whatever application built on top) to handle manually this logic for example to dispatch segments across multiple instances? If no, is it planned?

Reply:
This method was inteoduced in `feat/continuations` for testing purposes (along with `simulate_all_segments_interpreter`). The idea would rather be to use `prove`, which now takes some additional segment data as inputs to prove individual segments.